cmd/query/app/handler.go

```diff
@@ -399,24 +399,32 @@ func (aH *APIHandler) withTraceFromReader(
 	if !ok {
 		return
 	}
-	trace, err := reader.GetTrace(traceID)
+	trace, err := trace(traceID, reader, backupReader)
 	if err == spanstore.ErrTraceNotFound {
-		if backupReader == nil {
-			aH.handleError(w, err, http.StatusNotFound)
-			return
-		}
-		trace, err = backupReader.GetTrace(traceID)
-		if err == spanstore.ErrTraceNotFound {
-			aH.handleError(w, err, http.StatusNotFound)
-			return
-		}
+		aH.handleError(w, err, http.StatusNotFound)
+		return
 	}
 	if aH.handleError(w, err, http.StatusInternalServerError) {
 		return
 	}
 	process(trace)
 }
 
+func trace(
```

Comment:
every combination of `getTrace`, `getTraceFromReaders`, etc. was taken. Uh `getTraceFromReadersInternal`?